### PR TITLE
backend/repository2: set auth headers for getLayer

### PR DESCRIPTION
**Pull request by iaguis**
It was causing authenticated docker fetches to fail.

### CODE REVIEW

File: lib/backend/repository/repository2.go [hunk 1]
```diff
@@ -198,6 +198,10 @@ func (rb *RepositoryBackend) getLayerV2(layerID string, dockerURL *types.ParsedD
 		return nil, err
 	}
 
+	if rb.username != "" && rb.password != "" {
+		req.SetBasicAuth(rb.username, rb.password)
+	}
+
```
Comment (alban) [inline lib/backend/repository/repository2.go]: Is it the equivalent of `setAuthTokenV1` in `getLayerV1()`?

Comment (iaguis) [inline lib/backend/repository/repository2.go]: Yes. IIRC, in docker v1 you authenticate against the index, receive a token and then use that to query the registry. In docker v2, there's no distinction between index and registry, so you always use authentication.

Comment (alban) [inline lib/backend/repository/repository2.go]: This pattern is now used 4 times in this file. Maybe factorize:
```
rb.setBasicAuth(req)
```

**Comment by jonboulle** [comment]
Lgtm if  it works!

**Pull request merged**